docker/run.sh

```diff
@@ -0,0 +1,10 @@
+# For systems without journald
+mkdir -p /var/log/journal
+
+# Copy host libsystemd into image to avoid compatibility issues.
```

Comment:
Is this something that will work in all cases? Are there caveats?

Reply:
This is exactly what the fluentd-gcp team has had in place, so I'm hoping that it has worked for all of the cases that they've been exposed to.

docker run -it gcr.io/google-containers/fluentd-gcp:2.0.2 cat run.sh

You'll see the exact contents minus the copyrights up at the top. Not sure if we should add that back in or adjust it for our team.

Reply:
The cases that they've been exposed to were mostly Kubernetes nodes. Our container will likely need to run in other contexts, e.g., in raw Docker on various distros. Would it make sense to guard this behavior by an environment variable and only enable it for Kubernetes (or, more generally, for cases where users want systemd)?

Reply:
I feel that the case is already guarded by the check for the existence of /host/lib/libsystemd. I've validated this works in the docker context with the Flex team. The important element here is that it requires a mount such as -v /usr/lib64/:/host/lib/, this command will not get executed unless the host's lib dir is mounted in that way.

Reply:
Let's document that in the comment?

Reply:
Added, PTAL.